replication/binlogsyncer.go

```diff
@@ -178,6 +178,32 @@ func (b *BinlogSyncer) close() {
 		b.c.SetReadDeadline(time.Now().Add(100 * time.Millisecond))
 	}
 
+	// kill last connection id
+	if b.lastConnectionID > 0 {
+		// Use a new connection to kill the binlog dump
+		addr := ""
+		if strings.Contains(b.cfg.Host, "/") {
```

Comment:
I think you can abstract connect function so https://github.com/siddontang/go-mysql/blob/adefe8bbf3307d14ba1901c4f497fd4f950b5630/replication/binlogsyncer.go#L231 and here can use it.

Reply:
Fixed